seed/building_sync/building_sync.py

```diff
@@ -5,230 +5,139 @@
-                "key_path_value": "Custom ID 1",
-                "value_path_name": "auc:IdentifierValue",
-                "required": False,
-                "type": "string",
-            },
-            "gross_floor_area": {
-                "path": "auc:Sites.auc:Site.auc:Buildings.auc:Building.auc:FloorAreas.auc:FloorArea",
-                "key_path_name": "auc:FloorAreaType",
-                "key_path_value": "Gross",
-                "value_path_name": "auc:FloorAreaValue",
-                "required": True,
-                "type": "double",
-            },
-            "net_floor_area": {
-                "path": "auc:Sites.auc:Site.auc:Buildings.auc:Building.auc:FloorAreas.auc:FloorArea",
-                "key_path_name": "auc:FloorAreaType",
-                "key_path_value": "Net",
-                "value_path_name": "auc:FloorAreaValue",
-                "required": False,
-                "type": "double",
-            },
-            "footprint_floor_area": {
-                "path": "auc:Sites.auc:Site.auc:Buildings.auc:Building.auc:FloorAreas.auc:FloorArea",
-                "key_path_name": "auc:FloorAreaType",
-                "key_path_value": "Footprint",
-                "value_path_name": "auc:FloorAreaValue",
-                "required": False,
-                "type": "double",
-            },
-        }
+class BuildingSync(object):
+    BUILDINGSYNC_V2_PR1 = '2.0-pr1'
+    BUILDINGSYNC_V2_0 = '2.0'
+    VERSION_MAPPINGS_DICT = {
+        BUILDINGSYNC_V2_PR1: BASE_MAPPING_V2_PR1,
+        BUILDINGSYNC_V2_0: BASE_MAPPING_V2_0,
     }
 
     def __init__(self):
         self.filename = None
         self.data = None
         self.raw_data = {}
-
-    @property
-    def pretty_print(self):
-        """
-        Print the JSON to the screen
-        :return: None
-        """
-        print(json.dumps(self.raw_data, indent=2))
+        self.element_tree = None
+        self.version = None
 
     def import_file(self, filename):
         self.filename = filename
 
-        if os.path.isfile(filename):
-            with open(filename, 'rU') as xmlfile:
-                self.raw_data = xmltodict.parse(
-                    xmlfile.read(),
-                    process_namespaces=True,
-                    namespaces={
-                        'http://buildingsync.net/schemas/bedes-auc/2019': 'auc',
-                        'http://www.w3.org/2001/XMLSchema-instance': 'xsi',
-                    }
-                )
-
-        else:
+        if not os.path.isfile(filename):
             raise Exception("File not found: {}".format(filename))
 
+        # save element tree
+        with open(filename) as f:
+            self.element_tree = etree.parse(f)
+
+        # TODO: once xml translator has been implemented and used to convert
+        # files from 2.0-pr1 to 2.0, REMOVE the default argument so it fails
+        # if no proper version is found
+        self.version = self._parse_version(default=self.BUILDINGSYNC_V2_PR1)
```

Comment:
For now I'm allowing us to import files (either preexisting SEED bsync files _or_ newly uploaded files)  that don't have a schemaLocation that matches our expectations, and just uses v2.0-pr1 if it can't find one.

Just wanted to confirm that that's what we want to do for now until we are able to translate the files to 2.0.

Reply:
Decided that preexisting SEED files will be assumed to be 2.0-pr1 until the transformation is implemented. Also, any _newly_ imported files will fail to be imported if they don't have a parsable version.